runtime/src/main/java/org/corfudb/protocols/wireprotocol/TxResolutionInfo.java

```diff
@@ -46,6 +50,7 @@ public TxResolutionInfo(UUID txId, long snapshotTimestamp) {
         this.snapshotTimestamp = snapshotTimestamp;
         this.conflictSet = Collections.emptyMap();
         this.writeConflictParams = Collections.emptyMap();
+        this.poisonedStreams = Collections.emptySet();
```

Comment:
Cheaper serialization? Doubt it. The cost of Collections.emptySet is pretty close to just returning null (it returns **the** immutable empty set) and it keeps the code modular, as the implementation will just serialize a map with size 0. On the other hand we probably need to special case the deserializer when it gets a map/set of size 0, but that's a relatively easy fix.